demo/shaders/glsl/mesh_shadow_map.glsl

```diff
@@ -0,0 +1,40 @@
+// @[export]
+// @[internal_buffer]
+layout (set = 0, binding = 0) uniform PerViewData {
+    mat4 view;
+    mat4 view_proj;
+} per_view_data;
+
+// @[immutable_samplers([
+//     (
+//         mag_filter: Linear,
+//         min_filter: Linear,
+//         mip_map_mode: Linear,
+//         address_mode_u: Repeat,
+//         address_mode_v: Repeat,
+//         address_mode_w: Repeat,
+//         max_anisotropy: 16.0,
+//     )
+// ])]
+layout (set = 0, binding = 1) uniform sampler smp;
+
+// @[immutable_samplers([
+//     (
+//         mag_filter: Linear,
+//         min_filter: Linear,
+//         mip_map_mode: Linear,
+//         address_mode_u: ClampToEdge,
+//         address_mode_v: ClampToEdge,
+//         address_mode_w: ClampToEdge,
+//         anisotropy_enable: true,
+//         max_anisotropy: 16.0,
+//         compare_op: Greater,
+//     )
+// ])]
+layout (set = 0, binding = 2) uniform sampler smp_depth;
```

Comment:
I wondered if I should pull out these sampler definitions into a common file other than `mesh.glsl` so they could be imported into both `mesh.frag` and `mesh_shadow_map.vert`, but I wasn't sure if any code depended on these samplers being the same between both shaders. It seemed like the answer would be no.

Reply:
Yeah, I don't think we need any samplers when rendering shadow maps

Reply:
Removed.